modules/server_processes_manager/php/serverprocesslauncher.class.inc

```diff
@@ -39,10 +39,10 @@ class ServerProcessLauncher
     /**
      * Builds a new ServerProcessesLauncher
      *
-     * @param mixed $databaseProvider entity used to access the database
+     * @param IDatabaseProvider $databaseProvider entity used to access the database
      */
     public function __construct(
-        IDatabaseProvider $databaseProvider = null
```

Comment:
If the constructor is called with a null `databaseProvider`, why is `$this->_databaseProvider` not set to `new DefaultDatabaseProvider()` here instead of in the `get` function?

Reply:
You'd have to ask the person who wrote the code, I just fixed the bug that phan found. `null` isn't a valid value, because it's defined as taking an `IDatabaseProvider` in the signature, so it would crash with a runtime error if you called it that way anyways

Reply:
(the other problem that phan detected is that the doc comment said it was a "mixed" param type, but the actual signature in PHP was declared as IDatabaseProvider..)

Reply:
gotcha 👍